.gitignore

```diff
@@ -1,18 +1,145 @@
-/M2/Macaulay2/*/TAGS
-/M2/Macaulay2/m2/TAGS.doc
-/M2/build-vscode
+!/M2/BUILD/docker/storage/.emacs
+*.dSYM
+*.dep
+*.o
+*.out
+*.trim
+*~
+.#*
+.*.swp
+.DS_Store
+/M2/.completed
+/M2/.idea
+/M2/BUILD/CommonStagingArea
+/M2/BUILD/build/Macaulay2
```

Comment:
I think I should emphasize that combining literally every .gitignore is a bad idea, for several reasons:
1. if I make a copy of the `BUILD/build` directory, I want the .gitignore to be copied with it and stay local to that directory.
2. certain directory specific things are harder to maintain if they're clumped together in this file.
3. if we rename a directory, e.g. `M2/distributions`, we want the .gitignore to move with it.

Also, I personally think I prefer 20 .gitignores to a giant laundry list like this.

Reply:
Fair enough.  I'll revert that commit and just go back to the one with the `*~` files.  It was outside of the original scope of this PR anyway.